src/libstd/sync/once.rs

```diff
@@ -230,17 +230,48 @@ impl Once {
 
     /// Performs the same function as [`call_once`] except ignores poisoning.
     ///
-    /// [`call_once`]: struct.Once.html#method.call_once
-    ///
     /// If this `Once` has been poisoned (some initialization panicked) then
     /// this function will continue to attempt to call initialization functions
-    /// until one of them doesn't panic.
+    /// until one of them doesn't panic. Upon an initialization not panicking,
```

Comment:
"until one of them doesn't panic" feels weird to me. Does `call_once_force` continuously call its closure until it doesn't panic, or does it just suppress panics? That this paragraph references "initialization functions" also feels weird in isolation, since `call_once`/`call_once_panic` only takes one closure at a time.

Looking at the rest of the `Once` docs, it feels like a better explanation would involve calling out the states where the closure will and will not run: If the `Once` is uninitialized, then the closure will run regardless. If it's poisoned, then `call_once` will just panic, and `call_once_force` will run the closure. If the `Once` has successfully been initialized, then...? The closure will not run? The use of multiple initializers in the example leads me to the question of what happens when you hand a `Once` multiple `call_once` calls, with separate closures, if poisoning doesn't enter the picture.

Reply:
> Does call_once_force continuously call its closure until it doesn't panic, or does it just suppress panics?

if the `Once` hasn't been _successfully_ triggered (without panic), then `call_once_force` will call the closure _once_. otherwise, it will ignore the closure.

> If the Once has successfully been initialized, then...? The closure will not run?

yep, closure won't run

> The use of multiple initializers in the example leads me to the question of what happens when you hand a Once multiple call_once calls, with separate closures, if poisoning doesn't enter the picture.

if the first `call_once` succeeds, all following `call_once` and `call_once_force` calls are no-ops

Reply:
does https://github.com/rust-lang/rust/pull/45429/commits/d887138ddb73c998fca842e1b01f39d7745feede sound any clearer? thoughts?